jina/executors/indexers/vector.py

```diff
@@ -154,18 +157,31 @@ def raw_ndarray(self) -> Optional['np.ndarray']:
             return self._load_gzip(self.index_abspath)
         elif self.size is not None and os.stat(self.index_abspath).st_size:
             self.logger.success(f'memmap is enabled for {self.index_abspath}')
-            return np.memmap(self.index_abspath, dtype=self.dtype, mode='r', shape=(self.size, self.num_dim))
+            # `==` is required. `is False` does not work in np
+            deleted_keys = len(self.valid_indices[self.valid_indices == False])  # noqa
+            return np.memmap(self.index_abspath, dtype=self.dtype, mode='r',
+                             shape=(self.size + deleted_keys, self.num_dim))
 
     def query_by_id(self, ids: Union[List[int], 'np.ndarray'], *args, **kwargs) -> 'np.ndarray':
-        int_ids = [self.ext2int_id[j] for j in ids]
-        return self.raw_ndarray[int_ids]
+        """
+        Search the index by the external key (passed during `.add(`)
+
+        :param ids: The list of keys to be queried
+        """
+        # better naming. cannot change function signature for now
+        keys = ids
```

Comment:
I believe `deprecated_alias` is the decorator, that we have for exactly that use-case.

Reply:
You mean this? https://stackoverflow.com/a/49802489

Reply:
Did not know this, just saw the decorator earlier. Interesting read.